### PR TITLE
refactor!: add a dev-mode into the build flow

### PR DESCRIPTION
**Pull request by imkesin**
## Description

I have been extensively testing this locally... but I would appreciate at least one other person trying it on their machine. I would advise deleting the `/dist` directory so that there is an obvious clean slate.

### CODE REVIEW

File: bin/build.js [hunk 1]
```diff
@@ -1,43 +1,108 @@
...
+  return Promise.all(
+    [
+      rename(`${baseDir}/styles/index.css`, `${OUT_DIR}/index.css`),
+      withMap
+        ? rename(`${baseDir}/styles/index.css.map`, `${OUT_DIR}/index.css.map`)
+        : null
+    ].filter(Boolean)
+  )
+  .then(() => rm(`${baseDir}/styles`, { recursive: true }))
+}
```
Comment (doneel) [inline bin/build.js:64]: Am I understanding correctly that the way we can make it easier to import the CSS is to add to build step to literally just move the file?

File: bin/build.js [hunk 2]
```diff
@@ -1,43 +1,108 @@
...
+    ...esmConfig,
+    target: 'esnext',
+    minify: false,
+    sourcemap: true,
+    plugins: [
+      ...esmConfig.plugins,
+      {
+        name: 'rebuild-notify',
+        setup(build) {
+          build.onEnd(() => {
```
Comment (doneel) [inline bin/build.js:91]: Are the build settings/context different in this hot reload build versus the normal full build? I'm not sure I'm reading it correctly, I'm just looking for a way to reduce the differences between the two environments

File: src/components/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -310,7 +322,7 @@ export const DatePicker = ({
         onFocus={e => (e.target.readOnly = true)}
         onInputClick={() => {
           if (pickerRef.current && !isDesktop) {
-            pickerRef.current.setOpen(!pickerRef.current.isCalendarOpen)
+            pickerRef.current.setOpen(!pickerRef.current.isCalendarOpen())
```
Comment (doneel) [inline src/components/DatePicker/DatePicker.tsx:325]: How was this working before? Either it was broken or I'm missing some knowledge.

**Review by doneel — APPROVED** [review]
Looks exciting a couple clarification questions, but merge when you're ready.

File: bin/build.js (hunk 1 above)
Comment (imkesin) [reply bin/build.js:64]: Similar to our exported types, our CSS is not specific to ESM or CJS.  It happens to get initially nested inside a certain sub-folder, so I explicitly move it out of there.

The change in `package.json` (see `"exports" and "style"`) should make it so our css files end-up in a standard location.

File: src/components/DatePicker/DatePicker.tsx (hunk 3 above)
Comment (imkesin) [reply src/components/DatePicker/DatePicker.tsx:325]: The types for `react-datepicker` are awful. They only ported to a typescript approach in the latest version 7.X (and we're on 6.9 with an external types library).

It is actually a method, not a property, so it should be called.

This case can't even be reached on a mobile view due to some dark magic with the `onFocus` handler above ([reference to source code](https://github.com/Hacker0x01/react-datepicker/blob/02ce53f2c50ae77b177da1021b6e7f2e6faaaf0f/src/index.jsx#L484)). Still, the end-result is absolutely fine.

**Pull request merged**